c7n/filters/iamaccess.py

```diff
@@ -184,8 +184,8 @@ def normalize_condition(self, s):
         s_cond_op = list(s['Condition'].keys())[0]
 
         if s_cond_op not in conditions:
-            for s in set_conditions:
-                if not s_cond_op.startswith(s_cond_op):
+            for c in set_conditions:
+                if not s_cond_op.startswith(c):
```

Comment:
There is more wrong with this code block, still trying to figure out the correct behavior.

Reply:
Ok the code needs to change to say if `s_cond_op` does not start with at least one of the values in set_conditions, then return.  Otherwise if at least one of them matches, keep evaluating.

Reply:
Bah, I also wrote the ForAllValues wrong. Should be ForAllValues:StringEquals, not just ForAllValues...